Improve contribution guidelines

bastimeyer opened the pull request:
I originally just wanted to quickly add some missing stuff to the issue template, but I thought it would be better to have real contribution guidelines. I've copied the one that I'm currently using in the Twitch GUI repo and made a couple of changes. I also added a known issues file, which is currently just linking to the website, but could be improved later on, if desired.

I don't know whether this will help reducing the number of duplicates or whether it will increase the quality of issues, since people usually don't read stuff on the internet, but I think that it will at least not hurt having something more strict. If it is too strict or if there are parts that should be rewritten/reworded or if there are spelling mistakes, let me know or edit the PR. 😃

beardypig commented:
Sounds good to me, I don't have any concerns about it :-) 

There is another issues page in the docs (https://streamlink.github.io/issues.html), not sure why there are two... Perhaps we should merge those (see #119).

bastimeyer commented:
Isn't this more like a troubleshooting guide? I feel like there should be more stuff like verbose log outputs for debugging purposes or downloading streams for making sure that the player is responsible for certain issues and not streamlink (the stuff that gets asked a lot, you know).

gravyboat commented on CONTRIBUTING.md:
```diff
@@ -1,3 +1,99 @@
+
+5. **Format your issue report** — [well formatted texts][mastering-markdown] will make it much easier for developers to understand your report.
+
+Please try to be as detailed as possible in your report too. What is your environment? What steps will reproduce the issue? What would you expect to be the outcome? All these details will help people to assess and fix any potential bugs. The [issue template][issue-template] will aid you in structuring your report when submitting a new issue. If your report includes log outputs, please provide tho ...
```
Just a couple grammatical nitpicks on this line: `What would you expect to be the outcome` would make more sense as `What outcome do you expect` or `What would you expect the outcome to be`. Then for `If your report includes log outputs,` that makes more sense as `If your report includes log output`, so no need for the s, which then removes the need for the comma.

gravyboat commented on CONTRIBUTING.md:
```diff
@@ -1,3 +1,99 @@
+
+## Feature requests
+
+Feature requests are welcome. But take a moment to find out whether your idea fits with the scope and aims of the project. It's up to *you* to make a strong case to convince the project's developers of the merits of this feature. Please provide as much detail and context as possible.
```
Couple minor ones here just related to sentence flow: `Feature requests are welcome. But take a moment to find out whether your idea fits with the scope and aim of the project.` minor change to `Feature requests are welcome, but take a moment to find out whether your idea fits with the scope and aims of the project.`

gravyboat commented on CONTRIBUTING.md:
```diff
@@ -1,3 +1,99 @@
+
+## Plugin requests
+
+Plugin submissions and requests are a great way to improve Streamlink. Requests should be as detailed as possible and dedicated to only a single streaming service. Informations about the service as well as explicit URLs for live streams or recordings are required, so that the requested plugin can be easily implemented. Please be aware that plugins for streaming services that are using DRM protect ...
```
I don't think we need the `s` on `Informations` here. Minor change here to make it sound less demanding so we don't intimidate people on accident: `recordings are encouraged so that the requested plugin can be easily implemented.`

gravyboat commented on CONTRIBUTING.md:
```diff
@@ -1,3 +1,99 @@
+
+## Pull requests
+
+Good pull requests - patches, improvements, new features - are a fantastic help. They should remain focused in scope and avoid containing unrelated commits.
```
`Good pull requests - patches, improvements, new features` to `Good pull requests - patches, improvements, and new features` because I thing and belongs there, this one is mostly a suggestion and others may have a different view

gravyboat commented on ISSUE_TEMPLATE.md:
```diff
@@ -2,16 +2,17 @@
 - [ ] This is a feature request.
-- [ ] I used the search function to find already opened/closed issues or pull requests.
+- [ ] This is a plugin (improvement) request.
+- [ ] I have read the contribution guidelines.
```
Let's link to the contribution guidelines here so people don't have to search for them.

gravyboat commented on ISSUE_TEMPLATE.md:
```diff
@@ -21,13 +22,17 @@
-### Environment details (operating system, python version, etc.)
+### Environment details
+
+Operating system: ...
```
Operating system version would be a good addition in case people are using XP or something.

gravyboat commented on ISSUE_TEMPLATE.md:
```diff
@@ -2,16 +2,17 @@
 *Please read the contribution guidelines first!*
```
Let's add a note here about known issues and maybe link to them. I know this is redundant if you read the contribution guidelines, but people may not do that, so the lower we can make the barrier to entry the better. Maybe link to the contribution guidelines again as well.

bastimeyer replied:
The guidelines are already linked above the text area. I'm not sure if adding markdown links would benefit the readability of the template header. A quick note regarding known issues is reasonable though...

bastimeyer commented:
I have updated the PR. I'm not sure though if it makes sense using markdown links or even simple URLs in the issue template. The guidelines are already linked above the issue text area. That's why I have added a simple "see above" note. Some people are already failing at formatting the checklist or don't even understand the purpose of the template. Having links would make it much much harder to read and understand.

*Why is appveyor so slow...* 🤢

gravyboat commented:
Changes look good to me! I'll leave this open for a bit in case anyone else wants to contribute. Regardding appveyor I don't usually worry about it on these PRs, and yes it is extremely slow. I think they're just overloaded in terms of the queue. Travis has really spoiled me when it comes to fast checks.

gravyboat commented:
No feedback from anyone else so I'm going to merge this, thanks @bastimeyer!
